cleverhans/model.py

```diff
@@ -28,9 +29,15 @@ def get_layer(self, x, layer):
         :param x: A symbolic representation of the network input
         :param layer: The name of the hidden layer to return features at.
         :return: A symbolic representation of the hidden features
+        :raise: NoSuchLayerError if `layer` is not in the model.
         """
         # Return the symbolic representation for this layer.
-        return self.fprop(x)[layer]
+        output = self.fprop(x)
+        try:
+            requested = output[layer]
+        except KeyError:
+            raise NoSuchLayerError()
```

Comment:
try/catch is usually slower than the extra access, but if you are worried about that then you can do `requested = output.get(layer)` and it will be None if it doesn't exist.